lib/remote/jsonrpcconnection-pki.cpp

```diff
@@ -29,15 +29,18 @@ Value RequestCertificateHandler(const MessageOrigin::Ptr& origin, const Dictiona
 	String certText = params->Get("cert_request");
 
 	std::shared_ptr<X509> cert;
+	STACK_OF(X509) *chain;
 
 	Dictionary::Ptr result = new Dictionary();
 
 	/* Use the presented client certificate if not provided. */
 	if (certText.IsEmpty()) {
 		auto stream (origin->FromClient->GetStream());
 		cert = stream->next_layer().GetPeerCertificate();
+		chain = stream->next_layer().GetPeerCertificateChain();
```

Comment:
And these two together...

Reply:
As said, this will collect all X509-certificates the agent has put on the wire (which is already the case today). So to make this work only the master needs to collect this chain and use it in its verification function. Therefore it is only necessary to update the code on the masters-side, while the agent can be unchanged.